tests/utils/profilehandler/profile_handler.go

```diff
@@ -681,3 +681,157 @@ func CreateClusterByProfile(profile *Profile, client *rosacli.Client, waitForClu
+	if errDeleteCluster != nil {
+		if strings.Contains(output.String(), fmt.Sprintf("There is no cluster with identifier or name '%s'", cd.ClusterID)) {
+			log.Logger.Infof("Cluster %s is not existed.", cd.ClusterID)
+		} else {
+			log.Logger.Errorf("Error happened when delete cluster: %s", output.String())
+			errors = append(errors, errDeleteCluster)
+			return
+		}
+	} else {
+		if waitForClusterUninstall {
+			log.Logger.Infof("Waiting for the cluster %s to uninstalled", cd.ClusterID)
+			err = WaitForClusterUninstalled(client, cd.ClusterID, config.Test.GlobalENV.ClusterWaitingTime)
+			if err != nil {
+				log.Logger.Errorf("Error happened when waitling cluster uninstall: %s", err.Error())
+				errors = append(errors, err)
+			} else {
+				log.Logger.Infof("Delete cluster %s successfully.", cd.ClusterID)
+			}
+		}
+	}
+	// schedule KMS key
+	if ud.KMSKey != "" {
+		log.Logger.Infof("kms key: %s", ud.KMSKey)
+		err = ScheduleKMSDesiable(ud.KMSKey, region)
+		if err != nil {
+			log.Logger.Errorf("Error happened when schedule kms key: %s", err.Error())
+			errors = append(errors, err)
+		} else {
+			log.Logger.Infof("Schedule kms key successfully for cluster: %s", cd.ClusterID)
+		}
+	}
+	// schedule Etcd KMS key
+	if ud.EtcdKMSKey != "" {
+		log.Logger.Infof("etcd kms key: %s", ud.EtcdKMSKey)
+		err = ScheduleKMSDesiable(ud.EtcdKMSKey, region)
+		if err != nil {
+			log.Logger.Errorf("Error happened when schedule etcd kms key: %s", err.Error())
+			errors = append(errors, err)
+		} else {
+			log.Logger.Infof("Schedule etcd kms key successfully for cluster: %s", cd.ClusterID)
+		}
+	}
+	// delete audit log arn
+	if ud.AuditLogArn != "" {
+		log.Logger.Infof("audit log arn: %s", ud.AuditLogArn)
+		roleName := strings.Split(ud.AuditLogArn, "/")[1]
+		err = DeleteAuditLogRoleArn(roleName, region)
+		if err != nil {
+			log.Logger.Errorf("Error happened when delete audit log arn: %s", err.Error())
+			errors = append(errors, err)
+		} else {
+			log.Logger.Infof("Delete audit log arn successfully for cluster: %s", cd.ClusterID)
+		}
+	}
+	// delete vpc chain
+	if ud.VpcID != "" {
+		log.Logger.Infof("vpc id: %s", ud.VpcID)
+		err = DeleteVPCChain(ud.VpcID, region)
+		if err != nil {
+			log.Logger.Errorf("Error happened when delete vpc chain: %s", err.Error())
+			errors = append(errors, err)
+		} else {
+			log.Logger.Infof("Delete vpc chain successfully for cluster: %s", cd.ClusterID)
+		}
+	}
+	// delete operator roles
+	if ud.OperatorRolesPrefix != "" {
+		log.Logger.Infof("operator role prefix: %s", ud.OperatorRolesPrefix)
+		_, err := ocmResourceService.DeleteOperatorRoles("--prefix", ud.OperatorRolesPrefix, "--mode", "auto", "-y")
+		if err != nil {
+			log.Logger.Errorf("Error happened when delete operator role: %s", err.Error())
+			errors = append(errors, err)
+		} else {
+			log.Logger.Infof("Delete operator roles successfully for cluster: %s", cd.ClusterID)
+		}
+	}
+	// delete oidc config id
```

Comment:
Better to add another deletion about  if clusterID is not empty and oidcConfig=="" and cluster is sts cluster then run command `rosa delete oidc-provider -c <cluster> --mode auto -y` after cluster deletion finished.